src/widgets/images/utils/getResponse.ts

```diff
@@ -0,0 +1,49 @@
+import {getImages} from 'src/api';
+import {errorMessage} from 'src/constants';
+import {isFirstRender} from '../constants';
+import {HitsType} from 'src/api/imagesApi/typings';
+
+export const getResponse = async (
+  city: string,
+  country: string,
+  portion: string,
+  isInvalidCity: boolean,
+  invalidCountry: string,
+  isInvalidCountry: boolean,
+  setError: (value: string) => void,
+  isInvalidMasonryViewMode: boolean,
+  isInvalidCarouselViewMode: boolean,
+  setPhotosInState: (response: HitsType[]) => void,
+) => {
+  const isInvalidValue = (
+    (isInvalidMasonryViewMode && isInvalidCarouselViewMode) ||
+    ((!city || isInvalidCity) && (!country || isInvalidCountry))
+  );
+
+  if (isInvalidValue) {
+    return setError(errorMessage);
+  }
+
+  if (+portion < 3) {
+    return setError(errorMessage);
+  }
+
+  if (city && country && !isInvalidCountry) {
+    const result = await getImages(city, country, isFirstRender, portion);
+
+    if (result.data.hits.length === 0) {
+      const result = await getImages(city, country, isFirstRender, portion);
+      await setPhotosInState(result.data.hits);
+    }
+
+    return setPhotosInState(result.data.hits);
+  }
+
+  if (country) {
+    const result = await getImages(city, country, isFirstRender, portion);
+    await setPhotosInState(result.data.hits);
+  }
+
+  return setPhotosInState([]);
+};
```

Comment:
![image](https://user-images.githubusercontent.com/66788169/116362753-124bdb80-a80b-11eb-8316-24f2cdf06a0d.png)
what is the difference?? lines 35-36 vs 43-44